### PR TITLE
Cast factory test + cast factory.js fix

### PR DESCRIPTION
**Pull request by munkk**

### CODE REVIEW

File: src/core/expression/cast.factory.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {AppError} from '../infrastructure';
+import * as Convert from '../services/convert';
 
 export function castFactory(r) {
 	const rt = typeof r,
```
Comment (klumba12) [inline src/core/expression/cast.factory.js]: there is sence to use Convert.getType here too

Comment (vkorolev) [reply src/core/expression/cast.factory.js]: Please, add also failing test which would be fixed by `Convert.getType`

File: src/core/expression/cast.factory.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		it('should convert "123" to number', () => {
+			let foo = CastFactory.castFactory('123');
+			let result = foo(1);
+			expect(result).to.equal(123);
+		});
+
+		it('should convert string to Date', () => {
+			let foo = CastFactory.castFactory('2015-05-05');
+			let result = foo(new Date(2017));
+			expect(result instanceof Date).to.equal(true);
```
Comment (vkorolev) [inline src/core/expression/cast.factory.spec.js]: Maybe `expect(result).to.be.an.instanceof(Date)` ?

File: src/core/expression/cast.factory.spec.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import * as CastFactory from './cast.factory';
+
+describe('CastFactory', () => {
+	describe('castFactory', () => {
+		it('should return 123', () => {
+			let foo = CastFactory.castFactory(123);
```
Comment (klumba12) [inline src/core/expression/cast.factory.spec.js:6]: add tests for boolean and for constructros like new Number(123)

File: src/core/expression/cast.factory.js [hunk 4]
```diff
@@ -1,29 +1,36 @@
 import {AppError} from '../infrastructure';
 
+function getType(type) {
+	return {}.toString.call(type).slice(8, -1);
```
Comment (klumba12) [inline src/core/expression/cast.factory.js]: instead of 8 please write 'object '].length or something simlar for readability

**Review by klumba12 — APPROVED** [review]

**Pull request merged**